### PR TITLE
Preparatory work for flexible TV merge 

### PR DESCRIPTION
**Pull request by misgeatgit**
Addressing  https://github.com/opencog/opencog/issues/1295

### CODE REVIEW

File: opencog/atomspace/CountTruthValue.cc [hunk 1]
```diff
@@ -34,20 +34,20 @@
 
 using namespace opencog;
 
-CountTruthValue::CountTruthValue(strength_t m, confidence_t n, count_t c)
+CountTruthValue::CountTruthValue(strength_t m, confidence_t n, count_t c):TruthValue(COUNT_TRUTH_VALUE)
```
Comment (linas) [inline opencog/atomspace/CountTruthValue.cc]: Please don't do this. This makes the TV larger than 8 bytes. There is no reason to store a compile time constant as if it were some dynamic, run-time variable.

File: opencog/atomspace/TruthValue.cc [hunk 2]
```diff
@@ -94,3 +94,13 @@ bool TruthValue::isDefaultTV() const
...
 }
+
+TruthValueType TruthValue::getType() const
+{
+	return tv_type_;
+}
+
+void TruthValue::setType(TruthValueType t)
+{
+	tv_type_ = t;
```
Comment (linas) [inline opencog/atomspace/TruthValue.cc]: This is deeply and fundmentally wrong.  The different C++ classes have different numbers or members (simpe has 2, count has 3) and different merge methods. The numbers all have different meanings in the different classes. Changing the type is just .. impossible.

**Comment by linas** [comment]
I can't merge this as written.  I have nothing against MergeOption (although I wish it was called MergeStyle).  However the change to the type is deeply flawed and must be removed.

File: opencog/atomspace/TruthValue.cc (hunk 2 above)
Comment (misgeatgit) [inline opencog/atomspace/TruthValue.cc]: Hmm, I don't have any opposition about the optimization issue, so this PR might be turned down for that reason.If I have understood you correctly, you are in favor of changing the signature of the merge method (besides the naming issue) and oppose every other changes introduced. Right?

Comment (linas) [inline opencog/atomspace/TruthValue.cc]: Yes!  (and change the name to MergeStyle or something like that)

Comment (linas) [inline opencog/atomspace/TruthValue.cc]: BTW, I added Fuzzy and Probabilistic just a few days ago, for exactly this reason. The merge funcs for fuzzy and probabilistic are completely wrong, and need to be replaced by the correct formulas from the pln book.  The guts of SimpleTV merge func is wrong, as well.   Best to leave CountTV alone for now; if you want to change CountTV, then use either Probabilistic, or create a new TV, say, "FrequencyTruthValue"

**Pull request closed without merging**